src/core/Group.cpp

```diff
@@ -577,30 +577,36 @@ Entry* Group::findEntryByPath(QString entryPath, QString basePath)
     return nullptr;
 }
 
-Group* Group::findGroupByPath(QString groupPath, QString basePath)
+Group* Group::findGroupByPath(QString groupPath)
 {
-
     Q_ASSERT(!groupPath.isNull());
 
-    QStringList possiblePaths;
-    possiblePaths << groupPath;
-    if (!groupPath.startsWith("/")) {
-        possiblePaths << QString("/" + groupPath);
-    }
-    if (!groupPath.endsWith("/")) {
-        possiblePaths << QString(groupPath + "/");
-    }
-    if (!groupPath.endsWith("/") && !groupPath.endsWith("/")) {
-        possiblePaths << QString("/" + groupPath + "/");
+    // normalize the groupPath by adding missing front and rear slashes. once.
+    QString normalizedGroupPath;
+
+    if (groupPath == "") {
+        normalizedGroupPath = QString("/"); // root group
+    } else {
+        normalizedGroupPath = ((groupPath.startsWith("/"))? "" : "/")
+            + groupPath
+            + ((groupPath.endsWith("/") )? "" : "/");
     }
+    return findGroupByPathRecursion(normalizedGroupPath, "/");
```

Comment:
Why not just make the findGroupByPath function recursive?

Reply:
My thinkin was that the public entrypoint findGroupByPath does input normalization, but once normalized, the internal recursive step doesn't need to redo that normalization.

Reply:
Gotcha